### PR TITLE
Support connections to Sentinel and Cluster

### PR DESCRIPTION
**Pull request by junmipan**

### CODE REVIEW

**Comment by lukasmacko** [comment]
Did you update dependencies using `make update-dep`? 

There are files at `vendor/github.com/coreos/etcd/vendor/*`,  they should not be present if the glide is run with `--strip-vendor` flag.

Please rerun the update of dependencies using the make target mentioned above.

File: db/keyval/redis/bytes_broker_impl.go [hunk 1]
```diff
@@ -63,118 +62,110 @@ type bytesKeyVal struct {
...
 func (db *BytesConnectionRedis) Close() error {
 	if db.closed {
+		db.Debug("Close() called on a closed connection")
 		return nil
 	}
 	db.Debug("Close()")
 	db.closed = true
 	close(db.closeCh)
-	var buf bytes.Buffer
 	if db.pool != nil {
```
Comment (jozef-slezak) [inline db/keyval/redis/bytes_broker_impl.go]: Please use ligato utility/safeclose

File: db/keyval/redis/bytes_broker_impl.go [hunk 2]
```diff
@@ -187,169 +178,87 @@ func (db *BytesConnectionRedis) ListValues(match string) (keyval.BytesKeyValIter
 	return &bytesKeyValIterator{values: kvs}, nil
```
Comment (jozef-slezak) [inline db/keyval/redis/bytes_broker_impl.go]: is it possible to improve the implementation in term of cursor? (is it different story?)

File: db/keyval/redis/bytes_broker_impl.go (hunk 1 above)
Comment (junmipan) [inline db/keyval/redis/bytes_broker_impl.go]: Like this?
safeclose.Close(db.closeCh)

File: db/keyval/redis/bytes_broker_impl.go (hunk 2 above)
Comment (junmipan) [inline db/keyval/redis/bytes_broker_impl.go]: Yes, I will implement paging once the Iterator interface is updated (I think this update will impact etcd too)

File: db/keyval/redis/connection.go [hunk 3]
```diff
@@ -130,15 +69,310 @@ func createTLSConfig(config TLS) (*tls.Config, error) {
...
+	*/
+
+	// Declare these additional methods to enable access to them through this interface
+	Close() error
+	TxPipeline() goredis.Pipeliner
+	PSubscribe(channels ...string) *goredis.PubSub
+}
+
+// ClientConfig Configuration common to all types of Redis clients
+type ClientConfig struct {
```
Comment (jozef-slezak) [inline db/keyval/redis/connection.go]: Could you please move config to separate file config.go (like Lukas did in etcdv3). I have asked for that also Madhura (we could have a convention like this).

**Pull request merged**